catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/sharing/ShareableMetacardImpl.java

```diff
@@ -0,0 +1,180 @@
+    super(
+        new MetacardTypeImpl(
+            "metacard.sharing", Arrays.asList(new CoreAttributes(), new SecurityAttributes())));
+  }
+
+  public ShareableMetacardImpl(String id) {
+    this();
+    setId(id);
+  }
+
+  /**
+   * Check if a given metacard is a shareable metacard by checking the tags metacard attribute.
+   *
+   * @param metacard the metacard to check.
+   * @return true if the provided metacard is a shareable metacard, false otherwise.
+   */
+  public static boolean canShare(Metacard metacard) {
+    return metacard != null
+        && metacard
+            .getMetacardType()
+            .getAttributeDescriptors()
+            .stream()
+            .map(AttributeDescriptor::getName)
+            .anyMatch(attr -> Objects.equals(SecurityAttributes.ACCESS_GROUPS, attr))
+        && metacard
+            .getMetacardType()
+            .getAttributeDescriptors()
+            .stream()
+            .map(AttributeDescriptor::getName)
+            .anyMatch(attr -> Objects.equals(SecurityAttributes.ACCESS_INDIVIDUALS, attr));
+  }
+
+  /** Wrap any metacard as a ShareableMetacardImpl. */
+  public static ShareableMetacardImpl from(Metacard metacard) {
+    return new ShareableMetacardImpl(metacard);
+  }
+
+  public static ShareableMetacardImpl from(Map<String, Serializable> attributes) {
+    ShareableMetacardImpl shareableMetaCard = new ShareableMetacardImpl();
+
+    attributes
+        .entrySet()
+        .stream()
+        .forEach(entry -> shareableMetaCard.setAttribute(entry.getKey(), entry.getValue()));
+
+    return shareableMetaCard;
+  }
+
+  /**
+   * Get a copy of a shareable metacard.
+   *
+   * @param metacard
+   * @return
+   */
+  public static ShareableMetacardImpl clone(Metacard metacard) {
+    ShareableMetacardImpl shareableMetacard = new ShareableMetacardImpl();
+
+    metacard
+        .getMetacardType()
+        .getAttributeDescriptors()
+        .stream()
+        .forEach(
+            descriptor ->
+                shareableMetacard.setAttribute(metacard.getAttribute(descriptor.getName())));
+
+    return shareableMetacard;
+  }
+
+  /**
+   * Compute the symmetric difference between the access individuals
+   *
+   * @param metacard - metacard to diff against
+   */
+  public Set<String> diffSharingAccessIndividuals(Metacard metacard) {
+    if (canShare(metacard)) {
+      return Sets.symmetricDifference(
+          getAccessIndividuals(), from(metacard).getAccessIndividuals());
+    }
+    return Collections.emptySet();
+  }
+
+  /**
+   * Compute the symmetric difference between the access groups
+   *
+   * @param metacard - metacard to diff against
+   */
+  public Set<String> diffSharingAccessGroups(Metacard metacard) {
+    if (canShare(metacard)) {
+      return Sets.symmetricDifference(getAccessGroups(), from(metacard).getAccessGroups());
+    }
+    return Collections.emptySet();
+  }
+
+  protected List<String> getValues(String attributeName) {
```

Comment:
✏️ This is kind of a weird method. The only thing it gains over `AttributeImpl` is avoiding a null check for non-existent attributes elsewhere. It makes the business logic more opaque though by pushing that check down. Maybe we should rename it to something more descriptive. `getValuesOrEmpty`? `getValuesIfExists`?